accounts-db/src/accounts.rs

```diff
@@ -552,31 +569,56 @@ impl Accounts {
         account_locks: &mut AccountLocks,
         writable_keys: Vec<&Pubkey>,
         readonly_keys: Vec<&Pubkey>,
-    ) -> Result<()> {
+        entry_accts_lookup: &mut EntryAcctLocks,
+        allow_self_conflicting_entries: &bool,
+    ) -> (Result<()>, bool) {
+        let mut self_conflicting_account: bool = false;
+        let mut self_conflicting_tx = false;
         for k in writable_keys.iter() {
             if account_locks.is_locked_write(k) || account_locks.is_locked_readonly(k) {
-                debug!("Writable account in use: {:?}", k);
-                return Err(TransactionError::AccountInUse);
+                if *allow_self_conflicting_entries {
+                    self_conflicting_account = entry_accts_lookup.writables.contains(k)
+                        || entry_accts_lookup.readables.contains(k);
+                }
+                if !self_conflicting_account {
+                    debug!("Writable account in use: {:?}", k);
+                    return (Err(TransactionError::AccountInUse), self_conflicting_tx);
+                } else {
+                    if !self_conflicting_tx {
+                        self_conflicting_tx = true;
+                    }
```

Comment:
The `if` doesn't really accomplish anything here
```suggestion
                    self_conflicting_tx = true;
```

Reply:
addressed this and above points in the commit [e94bdeb429c2f66acae9669500cbb59ee30f35c0]